src/operator/timeout.ts

```diff
@@ -20,7 +20,8 @@ export function timeout<T>(this: Observable<T>, due: number | Date,
                            scheduler: Scheduler = async): Observable<T> {
   const absoluteTimeout = isDate(due);
   const waitFor = absoluteTimeout ? (+due - scheduler.now()) : Math.abs(<number>due);
-  const error = errorToSend || new TimeoutError();
+  const error = errorToSend instanceof Error ? errorToSend : new TimeoutError(errorToSend);
```

Comment:
I agree. It's kind of compromise I could think of without deep consideration. In here, I think requirement to achieve are
- allow pass custom err, either string or other object
- wrap it as `error`, to have stack trace

So in case of caller deliver its own error object directly, it won't generate stack trace as expected. Any suggestion to solve both of those?